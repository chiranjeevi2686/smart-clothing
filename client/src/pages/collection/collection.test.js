import React from 'react';
import { render } from '@testing-library/react';
import CollectionPage from './collection.component';
import { Provider } from 'react-redux';
import configureStore from 'redux-mock-store';

const defaultStore = {
    shop: {
        collections: {
            "hats": {
                "routeName": "hats",
                "id": "lQ1JWe8M0dlc5RPcGNDE",
                "title": "Hats",
                "items": [
                    {
                        "id": 1,
                        "price": 25,
                        "imageUrl": "https://i.ibb.co/ZYW3VTp/brown-brim.png",
                        "name": "Brown Brim"
                    },
                    {
                        "imageUrl": "https://i.ibb.co/ypkgK0X/blue-beanie.png",
                        "price": 18,
                        "name": "Blue Beanie",
                        "id": 2
                    },
                    {
                        "id": 3,
                        "name": "Brown Cowboy",
                        "imageUrl": "https://i.ibb.co/QdJwgmp/brown-cowboy.png",
                        "price": 35
                    },
                    {
                        "name": "Grey Brim",
                        "id": 4,
                        "imageUrl": "https://i.ibb.co/RjBLWxB/grey-brim.png",
                        "price": 25
                    },
                    {
                        "imageUrl": "https://i.ibb.co/YTjW3vF/green-beanie.png",
                        "name": "Green Beanie",
                        "price": 18,
                        "id": 5
                    },
                    {
                        "name": "Palm Tree Cap",
                        "price": 14,
                        "imageUrl": "https://i.ibb.co/rKBDvJX/palm-tree-cap.png",
                        "id": 6
                    },
                    {
                        "name": "Red Beanie",
                        "imageUrl": "https://i.ibb.co/bLB646Z/red-beanie.png",
                        "price": 18,
                        "id": 7
                    },
                    {
                        "imageUrl": "https://i.ibb.co/1f2nWMM/wolf-cap.png",
                        "id": 8,
                        "name": "Wolf Cap",
                        "price": 14
                    },
                    {
                        "name": "Blue Snapback",
                        "price": 16,
                        "imageUrl": "https://i.ibb.co/X2VJP2W/blue-snapback.png",
                        "id": 9
                    }
                ]
            }
        }}
};
const mockStore = configureStore();
jest.mock('react-router-dom', () => ({
  ...jest.requireActual('react-router-dom'),
  useParams: () => ({
    collectionId: 'hats',
  }),
}));

describe('CollectionPage component', () => {
    it('should render CollectionPage', () => {
        const { asFragment } = render(<Provider store={mockStore(defaultStore)}>
            <CollectionPage />
        </Provider>)
        expect(asFragment()).toMatchSnapshot();
    })
})

