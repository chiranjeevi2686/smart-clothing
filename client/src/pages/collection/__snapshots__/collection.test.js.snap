// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CollectionPage component should render CollectionPage 1`] = `
<DocumentFragment>
  <div
    class="sc-bdfBwQ cJFONO"
  >
    <h2
      class="sc-gsTCUz fBJmCA"
    >
      Hats
    </h2>
    <div
      class="sc-dlfnbm ejATpW"
    >
      <div
        class="sc-eCssSg bmGrxy"
      >
        <div
          class="sc-gKsewC cTsrRC"
        />
        <div
          class="sc-iBPRYJ imjZgE"
        >
          <span
            class="sc-fubCfw bUcFZH"
          >
            Brown Brim
          </span>
          <span
            class="sc-pFZIQ cXvWcx"
          >
            $25
          </span>
        </div>
        <button
          class="sc-hKgILt hXUWBl sc-jSgupP cwSpKa"
          data-test-id="add-cart-button"
        >
          ADD TO CART
        </button>
      </div>
      <div
        class="sc-eCssSg bmGrxy"
      >
        <div
          class="sc-gKsewC heyEiZ"
        />
        <div
          class="sc-iBPRYJ imjZgE"
        >
          <span
            class="sc-fubCfw bUcFZH"
          >
            Blue Beanie
          </span>
          <span
            class="sc-pFZIQ cXvWcx"
          >
            $18
          </span>
        </div>
        <button
          class="sc-hKgILt hXUWBl sc-jSgupP cwSpKa"
          data-test-id="add-cart-button"
        >
          ADD TO CART
        </button>
      </div>
      <div
        class="sc-eCssSg bmGrxy"
      >
        <div
          class="sc-gKsewC kyFpsC"
        />
        <div
          class="sc-iBPRYJ imjZgE"
        >
          <span
            class="sc-fubCfw bUcFZH"
          >
            Brown Cowboy
          </span>
          <span
            class="sc-pFZIQ cXvWcx"
          >
            $35
          </span>
        </div>
        <button
          class="sc-hKgILt hXUWBl sc-jSgupP cwSpKa"
          data-test-id="add-cart-button"
        >
          ADD TO CART
        </button>
      </div>
      <div
        class="sc-eCssSg bmGrxy"
      >
        <div
          class="sc-gKsewC huEZcX"
        />
        <div
          class="sc-iBPRYJ imjZgE"
        >
          <span
            class="sc-fubCfw bUcFZH"
          >
            Grey Brim
          </span>
          <span
            class="sc-pFZIQ cXvWcx"
          >
            $25
          </span>
        </div>
        <button
          class="sc-hKgILt hXUWBl sc-jSgupP cwSpKa"
          data-test-id="add-cart-button"
        >
          ADD TO CART
        </button>
      </div>
      <div
        class="sc-eCssSg bmGrxy"
      >
        <div
          class="sc-gKsewC jqrcol"
        />
        <div
          class="sc-iBPRYJ imjZgE"
        >
          <span
            class="sc-fubCfw bUcFZH"
          >
            Green Beanie
          </span>
          <span
            class="sc-pFZIQ cXvWcx"
          >
            $18
          </span>
        </div>
        <button
          class="sc-hKgILt hXUWBl sc-jSgupP cwSpKa"
          data-test-id="add-cart-button"
        >
          ADD TO CART
        </button>
      </div>
      <div
        class="sc-eCssSg bmGrxy"
      >
        <div
          class="sc-gKsewC gvswDz"
        />
        <div
          class="sc-iBPRYJ imjZgE"
        >
          <span
            class="sc-fubCfw bUcFZH"
          >
            Palm Tree Cap
          </span>
          <span
            class="sc-pFZIQ cXvWcx"
          >
            $14
          </span>
        </div>
        <button
          class="sc-hKgILt hXUWBl sc-jSgupP cwSpKa"
          data-test-id="add-cart-button"
        >
          ADD TO CART
        </button>
      </div>
      <div
        class="sc-eCssSg bmGrxy"
      >
        <div
          class="sc-gKsewC oOKWA"
        />
        <div
          class="sc-iBPRYJ imjZgE"
        >
          <span
            class="sc-fubCfw bUcFZH"
          >
            Red Beanie
          </span>
          <span
            class="sc-pFZIQ cXvWcx"
          >
            $18
          </span>
        </div>
        <button
          class="sc-hKgILt hXUWBl sc-jSgupP cwSpKa"
          data-test-id="add-cart-button"
        >
          ADD TO CART
        </button>
      </div>
      <div
        class="sc-eCssSg bmGrxy"
      >
        <div
          class="sc-gKsewC bvjWUB"
        />
        <div
          class="sc-iBPRYJ imjZgE"
        >
          <span
            class="sc-fubCfw bUcFZH"
          >
            Wolf Cap
          </span>
          <span
            class="sc-pFZIQ cXvWcx"
          >
            $14
          </span>
        </div>
        <button
          class="sc-hKgILt hXUWBl sc-jSgupP cwSpKa"
          data-test-id="add-cart-button"
        >
          ADD TO CART
        </button>
      </div>
      <div
        class="sc-eCssSg bmGrxy"
      >
        <div
          class="sc-gKsewC brUzlj"
        />
        <div
          class="sc-iBPRYJ imjZgE"
        >
          <span
            class="sc-fubCfw bUcFZH"
          >
            Blue Snapback
          </span>
          <span
            class="sc-pFZIQ cXvWcx"
          >
            $16
          </span>
        </div>
        <button
          class="sc-hKgILt hXUWBl sc-jSgupP cwSpKa"
          data-test-id="add-cart-button"
        >
          ADD TO CART
        </button>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
